### PR TITLE
[RFR][1LP] Extend DB migrate test, refactor appliance

### PR DESCRIPTION
**Pull request by jkrocil**
Enhance the DB migration test to be able to migrate bigger databases
Refactor/update utils.appliance
Update DB restore test

{{pytest: -k "test_db_migrate and scvmm_5620"}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 965b2061e133808389e8dd62b7300a0ab8acc27c:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b13ebca0d9ef198f30c90b13c196c5cbbfad77d7:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 907130ae29993edaf5c36a584fca7a055f7a7a43:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3e75811de297228954b59fa35363fdc62aade231:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by jkrocil** [comment]
Re PRT results: Tests are still running (and passing) but the results, even if some fail, are irrelevant. As long as they run, and they do.

**Comment by jkrocil** [comment]
Actually, one more thing...

**Comment by dajoRH** [comment]
Lint report for commit 0e4457216022aed75a2013603b0126e0f9b28d5c:

`cfme/tests/test_db_migrate.py`:
- :red_circle: Line 5:1: **F401** *'db' imported but unused*
- :red_circle: Line 10:1: **F401** *'wait_for' imported but unused*

`cfme/tests/test_db_restore.py`:
- :red_circle: Line 90:9: **F821** *undefined name 'aapl2'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 38efacfa8744da7691d498683a2f7845ed5db81d:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by jkrocil** [comment]
Ok, ready for review. I only let a single test run and only for 5.6 stream - the main issue with this PR was the fixture + cleanup which is now working well.

File: cfme/tests/test_db_migrate.py [hunk 1]
```diff
@@ -21,13 +20,40 @@ def pytest_generate_tests(metafunc):
     return metafunc.parametrize(argnames=argnames, argvalues=argvalues, ids=idlist)
 
 
+@pytest.fixture(scope="session")
+def extend_db_partition():
+    app = pytest.store.current_appliance
```
Comment (RonnyPfannschmidt) [inline cfme/tests/test_db_migrate.py]: please use the get_or_create function that backs the current_appliance context local in that place

File: cfme/tests/test_db_migrate.py [hunk 2]
```diff
@@ -21,13 +20,40 @@ def pytest_generate_tests(metafunc):
...
+    app.stop_evm_service()
+    app.extend_db_partition()
+    app.start_evm_service()
+
+
+@pytest.yield_fixture(scope="session")
+def backup_orig_state(extend_db_partition):
+    app = pytest.store.current_appliance
```
Comment (RonnyPfannschmidt) [inline cfme/tests/test_db_migrate.py]: dito

Comment (jkrocil) [inline cfme/tests/test_db_migrate.py]: Updated.

**Pull request merged**